bciers/apps/reporting/src/app/components/verification/VerificationForm.tsx

```diff
@@ -5,57 +5,254 @@ import MultiStepFormWithTaskList from "@bciers/components/form/MultiStepFormWith
+    const isVisitNamesArray = Array.isArray(updatedData.visit_names);
+
+    if (isVisitNamesArray) {
+      // LFO scenario
+      const selectedValues = updatedData.visit_names;
+
+      if (selectedValues.includes("None")) {
+        // If "None" is selected:
+        // - Lock selection to only "None"
+        updatedData.visit_names = ["None"];
+
+        // - Clear `visit_types` and `visit_others`
+        updatedData.visit_types = [];
+        updatedData.visit_others = [];
+      } else {
+        // If "None" is not selected:
+        updatedData.visit_names = selectedValues.filter(
+          (value: string) => value !== "None",
+        );
+
+        // Update `visit_types` for facilities except "Other"
+        updatedData.visit_types = updatedData.visit_names
+          .filter((visit_name: string) => visit_name !== "Other")
+          .map((visit_name: string) => {
+            const existingVisitType = updatedData.visit_types?.find(
+              (item: { visit_name: string }) => item.visit_name === visit_name,
+            );
+            return (
+              existingVisitType || {
+                visit_name,
+                visit_type: "",
+              }
+            );
+          });
+
+        // If "Other" is selected, prepare `visit_others`
+        if (selectedValues.includes("Other")) {
+          updatedData.visit_others = updatedData.visit_others ?? [
+            {
+              visit_name: "",
+              visit_coordinates: "",
+              visit_type: "",
+            },
+          ];
+        } else {
+          updatedData.visit_others = [{}];
+        }
+      }
+    } else {
+      // SFO scenario
+      const selectedVisitName = updatedData.visit_names;
+
+      if (selectedVisitName === "None") {
+        // If "None" is selected:
+        // - Lock selection to "None"
+        updatedData.visit_names = "None";
+
+        // - Clear `visit_types` and `visit_others`
+        updatedData.visit_types = null;
+        updatedData.visit_others = [{}];
+      } else if (selectedVisitName) {
+        // If a specific visit name is selected (e.g., "Facility X"):
+        if (selectedVisitName !== "Other") {
+          // Prepare or retain `visit_types`
+          updatedData.visit_types = updatedData.visit_types ?? {
+            visit_type: "",
+          };
+          // - Clear `visit_others`
+          updatedData.visit_others = [{}];
+        } else {
+          // If "Other" is selected, clear `visit_types`
+          updatedData.visit_types = null;
+
+          // Prepare `visit_others`
+          updatedData.visit_others = updatedData.visit_others ?? {
+            visit_name: "",
+            visit_coordinates: "",
+            visit_type: "",
+          };
+        }
+      } else {
+        // If no selection is made:
+        updatedData.visit_types = null;
+        updatedData.visit_others = [{}];
+      }
+    }
+
+    // 🔄 Update the form data state with the modified data
```

Comment:
This is a LOT of logic to run on every single key stroke on the form. I think it would be very helpful to try to build this logic into the schema itself instead